src/styleUtils.ts

```diff
@@ -0,0 +1,102 @@
+import {processColor, Platform} from 'react-native';
+import type * as MarkdownTextInputDecoractorView from './MarkdownTextInputDecoratorViewNativeComponent';
+
+type MarkdownStyle = MarkdownTextInputDecoractorView.MarkdownStyle;
+
+type PartialMarkdownStyle = Partial<{
+  [K in keyof MarkdownStyle]: Partial<MarkdownStyle[K]>;
+}>;
+const FONT_FAMILY_MONOSPACE = Platform.select({
+  ios: 'Courier',
+  default: 'monospace',
+});
+
+function makeDefaultMarkdownStyle(): MarkdownStyle {
+  return {
+    syntax: {
+      color: 'gray',
+    },
+    link: {
+      color: 'blue',
+    },
+    h1: {
+      fontSize: 25,
+    },
+    blockquote: {
+      borderColor: 'gray',
+      borderWidth: 6,
+      marginLeft: 6,
+      paddingLeft: 6,
+    },
+    code: {
+      fontFamily: FONT_FAMILY_MONOSPACE,
+      color: 'black',
+      backgroundColor: 'lightgray',
+    },
+    pre: {
+      fontFamily: FONT_FAMILY_MONOSPACE,
+      color: 'black',
+      backgroundColor: 'lightgray',
+    },
+    mentionHere: {
+      color: 'green',
+      backgroundColor: 'lime',
+    },
+    mentionUser: {
+      color: 'blue',
+      backgroundColor: 'cyan',
+    },
+  };
+}
+
+function mergeMarkdownStyleWithDefault(input: PartialMarkdownStyle | undefined, shouldAddUnitsToNumberValues = false): MarkdownStyle {
+  const output = makeDefaultMarkdownStyle();
+
+  if (input !== undefined) {
+    Object.keys(input).forEach((key) => {
+      if (!(key in output)) {
+        return;
+      }
+      Object.assign(output[key as keyof MarkdownStyle], input[key as keyof MarkdownStyle]);
+    });
+  }
+
+  if (shouldAddUnitsToNumberValues) {
```

Comment:
I don't like that this function does two things at once, can we move adding units to a separate function `processUnitsInMarkdownStyle`? Also, there should be some function from `react-native-web` that already does that (see how `processColor` is used below)